AdminWebsite/AdminWebsite/ClientApp/src/app/work-allocation/edit-work-hours/vho-work-hours-non-availability-table/vho-work-hours-non-availability-table.component.ts

```diff
@@ -39,12 +39,19 @@ export class VhoWorkHoursNonAvailabilityTableComponent implements OnInit, CanDea
         });
     }
     @Input() set result(value: VhoNonAvailabilityWorkHoursResponse[]) {
-        this.resetStartDateAndEndDate();
+        //if the filter by date form is not filtered, reset the start and end dates
+        if (!this.isFiltered) {
+            this.resetStartDateAndEndDate();
+        }
         this.hideMessage();
         if (value) {
             this.nonAvailabilityWorkHoursResponses = value;
             this.nonWorkHours = value.map(x => this.mapNonWorkingHoursToEditModel(x));
             this.nonWorkHours = this.nonWorkHours.slice(0, this.filterSize);
+            //if the form has has already been filtered, run-run the filter again
+            if (this.isFiltered) {
+                this.filterByDate();
```

Comment:
can we add a test to assert that the date filter is applied here? If you check the sonar cloud coverage, you'll see that this line isn't covered by tests

Reply:
@oliver-scott I've added a test that will cover these additional lines.